tapdb/sqlc/migrations/000027_better_universe_stats.up.sql

```diff
@@ -0,0 +1,37 @@
+DROP VIEW universe_stats;
+
+CREATE VIEW universe_stats AS
+WITH sync_counts AS (
+    SELECT universe_root_id, COUNT(*) AS count
+    FROM universe_events
+    WHERE event_type = 'SYNC'
+    GROUP BY universe_root_id
+), proof_counts AS (
+    SELECT universe_root_id, event_type, COUNT(*) AS count
+    FROM universe_events
+    WHERE event_type = 'NEW_PROOF'
+    GROUP BY universe_root_id, event_type
+), aggregated AS (
+    SELECT COALESCE(SUM(count), 0) as total_asset_syncs,
+           0 AS total_asset_proofs,
+           universe_root_id
+    FROM sync_counts
+    GROUP BY universe_root_id
+    UNION ALL
+    SELECT 0 AS total_asset_syncs,
+           COALESCE(SUM(count), 0) as total_asset_proofs,
+           universe_root_id
+    FROM proof_counts
+    GROUP BY universe_root_id
+)
+SELECT
+    SUM(ag.total_asset_syncs) AS total_asset_syncs,
+    SUM(ag.total_asset_proofs) AS total_asset_proofs,
+    roots.asset_id,
+    roots.group_key,
+    roots.proof_type
+FROM aggregated ag
+JOIN universe_roots roots
+    ON ag.universe_root_id = roots.id
+GROUP BY roots.asset_id, roots.group_key, roots.proof_type
```

Comment:
Should we not tweak the index here like in the test?
```
		CREATE INDEX IF NOT EXISTS idx_universe_leaves_asset
		ON universe_leaves(
			asset_genesis_id, universe_root_id
		);`
```

Reply:
Same with `idx_mssmt_nodes_composite`

Reply:
I'm not sure what you mean? Extend the indexes to cover the fields mentioned in `GROUP BY`? Those are all covered already be pre-existing indexes.

Reply:
No I just noticed that the names of some of the indexes that were removed in the test didn't actually exist under those names, that's the only thing changed in the indexes. All other commits are about having a more optimized version of the `universe_stats` view.

Reply:
Got it -- thanks for the clarification 👍